### DISCUSSION TITLE
Load testing application with locust

### DISCUSSION THREAD

**Discussion by veigajoao** [miguelgrinberg/Flask-SocketIO#1784] [Ideas]
Hey guys,

I just deployed a test version of my application and I am a little lost regarding load testing.

Is there any recommended framework for load testing flask-socketIO?

I usually use locust, but I am having a little trouble setting up the websocket test.

Also, as a suggestion to @miguelgrinberg, a testing section on the documentation would be very helpful!

**Comment by miguelgrinberg** [comment]
Does locust support Socket.IO? You can't use WebSocket for this, that's a different, much lower level protocol. When I needed to load test a Socket.IO app I just built a client using the python-socketio package. I would imagine there must be some load testing tools that support Socket.IO, but I don't recall seeing any myself.

**Reply by veigajoao** [reply]
Thanks for the reply @miguelgrinberg. I was hoping you had something to recommend, I'llbuild the client from the python-socketio module then. Locust doesn't support it out of the box.

I would still like ti suggest adding a testing section to the documentation, or at least some references, think that it would be very helpful to a lot of people (and also reduce the number of discussions here).

Thanks a lot!

**Reply by bluthen** [reply]
We've load tested with Gatling before. Used java socket.io client and had to write a custom action. It took me a long time actually to figure it out because I wasn't that familiar with scala or gatling. But that was the load testing cluster that was available to me.

https://github.com/socketio/socket.io-client-java


Locust has an example of sockets io but I wouldn't use that. I'm not sure why they didn't just use the python socketio client. It looks like they manually send the socket io protocol. But maybe you can take it and replace most of it with using the python socket io client.
https://github.com/SvenskaSpel/locust-plugins/blob/master/examples/socketio_ex.py
